extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -152,39 +152,32 @@ function () {
 	}
 
 	/**
-	 * Get all contributions for a user, limited by most recent n days if $days is not null
+	 * Get count of contributions for a user in the current week( Starting Sunday midnight )
 	 *
 	 * @param User $user
-	 * @param weekly If true get user's contributions current week. Otherwise for entire membership period (2 years)
 	 * @return int Number of contributions
 	 */
-	public function getUserContributions( User $user, $weekly = true ) {
+	public function getUserContributionsCount( User $user ) {
+
 		$userId = $user->getId();
 
 		$revisionCount = WikiaDataAccess::cache(
 			// TODO: Should purge this when user edits
-			wfMemcKey( self::CURR_USER_CONTRIBUTIONS_MCACHE_KEY, $userId, $weekly ),
+			wfMemcKey( self::CURR_USER_CONTRIBUTIONS_MCACHE_KEY, $userId ),
 			WikiaResponse::CACHE_VERY_SHORT, // short cache b/c it's for the current user's info
-			function () use ( $userId, $weekly ) {
+			function () use ( $userId ) {
 				$db = wfGetDB( DB_SLAVE );
 
-				if ( $weekly ) {
-					// From last Sunday (matches wikia_user_properties)
-					$dateFilter = 'rev_timestamp >= FROM_DAYS(TO_DAYS(CURDATE()) - MOD(TO_DAYS(CURDATE()) - 1, 7))';
-				} else {
-					$dateFilter = 'rev_timestamp > DATE_SUB(now(), INTERVAL 2 YEAR)';
-				}
+				$propertyName = 'editcountThisWeek';
 
 				$sqlData = ( new WikiaSQL() )
-					->SELECT( 'count(rev_id) AS revision_count' )
-					->FROM( 'revision' )
-					->WHERE( 'rev_user' )->EQUAL_TO( $userId )
-					->AND_( $dateFilter )
-					->runLoop( $db, function ( &$sqlData, $row ) {
-						$sqlData = $row->revision_count;
-					} );
+					->SELECT( 'wup_value' )
+					->FROM( 'wikia_user_properties' )
+					->WHERE( 'wup_user' )->EQUAL_TO( $userId )
+					->AND_( 'wup_property' )->EQUAL_TO( $propertyName )
+					->run( $db );
 
-				return $sqlData;
+				return $sqlData->fetchObject()->wup_value;
```

Comment:
`fetchObject()` may return false so we should check it before access to `wup_value` or you can use `runLoop` method and then just return the result